### PR TITLE
Convert whereBetween values to UTCDateTime

### PR DESCRIPTION
**Pull request by ricardofontanelli**
Added a protected method (dateTimeConvertion) to convert DateTime objects to MongoDB\BSON\UTCDateTime and added a verification for DateTime objects in $where['values'] array to ensure that all DateTime values will be properly converted to MongoDB\BSON\UTCDateTime.

### CODE REVIEW

File: src/Jenssegers/Mongodb/Query/Builder.php [hunk 1]
```diff
@@ -1018,6 +1029,20 @@ protected function compileWhereRaw($where)
...
+    /**
+     * Convert to MongoDB\BSON\UTCDateTime if $value is a DateTime object.
+     *
+     * @param  mixed   $value
+     * @return mixed
+     */
+    protected function dateTimeConvertion($value)
+    {
+        if ($value instanceof DateTime) {
+            return new UTCDateTime($value->getTimestamp() * 1000);
```
Comment (GreenLamantin) [inline src/Jenssegers/Mongodb/Query/Builder.php]: Why UTCDateTime instead UTCDatetime?

Comment (ricardofontanelli) [inline src/Jenssegers/Mongodb/Query/Builder.php]: According PHP [documentation](http://php.net/manual/pt_BR/class.mongodb-bson-utcdatetime.php), the class name is UTCDate**T**ime

Comment (aditya-cherukuri) [inline src/Jenssegers/Mongodb/Query/Builder.php]: Isn't the if conditional check redundant? You are already checking if the value is an instance of DateTime before calling the dateTimeConversion function.

**Comment by Opsse** [comment]
This issue isn't fixed yet ?

**Comment by ricardofontanelli** [comment]
It was merged at moloquent/moloquent, I don't know why nobody has merged this PR here!

**Comment by D3nnisH** [comment]
already fixed by #1328

**Comment by divine** [comment]
> already fixed by #1328
Then this PR should be closed.

**Pull request closed without merging**